personalrobotics/ada_ros2
# Configure `ada_description` for Articulable Fork

# Description

This pull request adds a configuration to specify the `end_effector_tool` which dynamically modifies the URDF from `ada.xacro`. This will allow users to specify tools from `none`, `fork`, and `articulable_fork` that will manifest kinematically and visually. For developers using ADA before the articulable fork hardware is transitioned in, set the tool to `fork`.

Related PRs: https://github.com/personalrobotics/ada_ros2/pull/53, https://github.com/personalrobotics/ada_ros2/pull/54, https://github.com/personalrobotics/ada_feeding/pull/211

# Testing procedure

- \[x\] Run `ros2 launch ada_description view_ada.launch.py end_effector_tool:=<tool>` for each `<tool>` in `none`, `fork`, and `articulable_fork`
- \[x\] Verify that ADA in RViz2 populates as expected with `none` having no tool, `fork` having the rigid fork assembly, and `articulable_fork` having the articulable fork whose joints can be commanded with the slider window

# Before opening a pull request

- \[x\] `pre-commit run --all-files`
- \[x\] Run your code through [pylint](https://pylint.readthedocs.io/en/latest/). `pylint --recursive=y --rcfile=.pylintrc .`. All warnings but `fixme` must be addressed.

# Before Merging
- \[ \] `Squash & Merge`
